File: python/easy/merge_strings_alternately.py
# 1768. Merge Strings Alternately
# https://leetcode.com/problems/merge-strings-alternately/

class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        max_length = max([len(word1), len(word2)])
        answer = []
        for i in range(0, max_length):
            if i < len(word1):
                answer.append(word1[i])
            if i < len(word2):
                answer.append(word2[i])
        return "".join(answer)

File: javascript/easy/merge_strings_alternately.js
// 1768. Merge Strings Alternately
// https://leetcode.com/problems/merge-strings-alternately/

var mergeAlternately = function(word1, word2) {
    var maxLength = Math.max(word1.length, word2.length);
    var answer = [];
    
    for (var i = 0; i < maxLength; i++) {
        if (i < word1.length) {
            answer.push(word1[i]);
        }
        
        if (i < word2.length) {
            answer.push(word2[i]);
        }
    }
    
    return answer.join("");
};